src/main/java/ui/windows/StatsController.java

```diff
@@ -17,39 +20,39 @@ public StatsController() {
 
     /**
      * Sets a display endpoint for the controller.
-     * @param display The display to be used.
+     * @param display the display to be used
      */
     public void setDisplay(StatsDisplay display) {
         this.display = display;
     }
 
     /**
      * Requests statistics for a given user and updates display.
-     * @param player the PlayerInformation of the user
      * @param username the name of the user
      * @return false if no display is supplied, true otherwise
      */
-    public boolean tryRequestUser(PlayerInformation player, String username) {
+    public boolean tryRequestUser(String username) throws IOException {
         if (display == null) {
             return false;
         }
-
+        CSVDatabase database = new CSVDatabase();
```

Comment:
ideally this CSVDatabase (or really any implementation of DataAccess) is passed in to the controller, e.g.

```
public class StatsController {
  private DataAccess database;
  // other attributes

  public StatsController(..., DataAccess database) {
    // set other attributes
    this.database = database
  }

  // other methods

  public boolean tryRequestUser( ... {
    // ...

    PlayerInformation player = database.loadPlayer(username);
    // rest of your code
  }
```

this way we can keep the controller separate from the implementation of the database, make them interchangeable, etc